api/src/main/java/com/graphhopper/GHRequest.java

```diff
@@ -34,6 +34,9 @@
  */
 public class GHRequest {
     private List<GHPoint> points;
+    // todonow: keep this here or put it into hints, and even more important: can we remove vehicle+weighting from
+    // hints?
+    private String profile = "";
```

Comment:
I think we discussed this in #1859 already: We'll def. remove vehicle+weighting from the HintsMap, but maybe in a follow-up PR? Since the profile parameter will be required it makes sense to keep it as a separate field?

Reply:
Yes, I would also use it here and remove weighting and vehicle in a later PR